### PR TITLE
Use PyTorch's dynamo benchmark skip-list.

### PR DESCRIPTION
**Pull request by ysiraichi**
This PR starts using PyTorch's dynamo benchmark skip-list for skipping specific models, based on the experiments' configuration. Besides that, it also:

- Refactors `is_compatible` function
- Refactors the lookup of a given file, previously in `add_torchbench_dir` function
- Replaces the comments on each skipped benchmark by its actual error message
    - I'd say it's more informative, but will revert it if anyone thinks otherwise 

cc @miladm

### CODE REVIEW

File: benchmarks/torchbench_model.py [hunk 1]
```diff
@@ -64,59 +65,51 @@
...
         "test": "train"
-    },],  # not implemented
+    },],  # Model's DEFAULT_TRAIN_BSIZE is not implemented
     "pyhpc_turbulent_kinetic_energy": [{
         "test": "train"
-    },],  # not implemented
-    "pytorch_struct": [{
-        "test": "eval"
-    },],  # not implemented
+    },],  # Model's DEFAULT_TRAIN_BSIZE is not implemented
```
Comment (zpcore) [inline benchmarks/torchbench_model.py:112]: Thanks for the refining. Does those models removed from the list work fine now?

Comment (ysiraichi) [reply benchmarks/torchbench_model.py:112]: Not really. But, as far as I can see, they are not in torchbench list of models anymore

**Review by zpcore — APPROVED** [review]

File: benchmarks/torchbench_model.py [hunk 2]
```diff
@@ -218,16 +230,29 @@ def is_compatible(self,
...
+    if name in self.skip["test"].get(benchmark_experiment.test, {}):
+      return False
+
+    if name in self.skip["device"].get(benchmark_experiment.accelerator, {}):
+      return False
+
+    if name in self.skip["multiprocess"]:
+      # No support for multiprocess, yet. So, skip all benchmarks that
+      # only work with it.
+      return False
```
Comment (miladm) [inline benchmarks/torchbench_model.py:248]: I wonder if we should include an "INFO" message when skipping models so it's clear on the logs - unless it gets reported elsewhere. wdyt @zpcore?

Comment (ysiraichi) [reply benchmarks/torchbench_model.py:248]: The scripts already report when a model is skipped. However, the reason is not logged. Do you think we should log that, too?

File: benchmarks/torchbench_model.py [hunk 3]
```diff
@@ -169,32 +154,59 @@ def __init__(self, args):
...
     return torchbench_dir
 
+  def get_skip_data(self):
+    """Retrieve the skip data in the PyTorch YAML file.
+
+    Reads the YAML file in PyTorch's dynamo benchmarks directory, and transform
+    its lists of models into sets of models.
+    """
+
+    benchmarks_dir = self._find_near_file(("benchmarks",))
```
Comment (zpcore) [inline benchmarks/torchbench_model.py:193]: It turns out pytorch/benchmarks and xla/benchmarks share the same `benchmarks` name. And when we search for yaml file, the nearest will be the xla/benchmarks.

**Pull request merged**